base/strings/basic.jl

```diff
@@ -630,6 +653,36 @@ end
     end
     return  _isascii(cu,last-chunk_size+1,last)
 end
+
+function isascii(bytes::AbstractVector{UInt8}, first, last)
```

Comment:
```suggestion
function _isascii_bytes_chunked(bytes::AbstractVector{UInt8}, first, last)
```

I think this is internal and not considered API. Renamed to not conflict with the `_isascii(::AbstractVector{CU})`  from above, but maybe that should get a clearer name as well?

Reply:
There are many places and algorithm that using a fast isascii can speed things up.  My thinking was that in this form it is as useful as the version that uses just an array, as it stops you from having to create a view or substring..

Reply:
I don't personally see the need for this, since creating a view for this is cheap, should not be different performance wise than passing a "proper" array (not to mention that `codeunits` already gives a view-like wrapper, with no repercussions for performance) as well as the other `isascii` already taking `AbstractVector{UInt8}`. 

If this is intended as new API, it needs at least docs and bounds checking for `first` and `last`.

Reply:
@oscardssmith, @StefanKarpinski, @stevengj do you think that it would be worth while to add this to the API?  I agree with @Seelengrab about the docs and the bounds checking. 

My thinking is that while yes you could create a `view` or a `SubString` to do the same thing, there is some overhead associated with that.  You can see it in the `@code_native` of this function when creating a view for the `reinterpret`.   So UTF-8 tasks can be sped up by checking for `isascii` this flexibility seems useful.